ValetTouchIDTest/ValetTouchIDTestViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  ValetTouchIDTestViewController.swift
+//  Valet
+//
+//  Created by Eric Muller on 4/20/16.
+//  Copyright © 2016 Square, Inc. All rights reserved.
+//
+
+import UIKit
+
+
+final class ValetTouchIDTestViewController : UIViewController
+{
+    // MARK: Properties
+
+    @IBOutlet var textView : UITextView?
+    let secureEnclaveValet : VALSecureEnclaveValet = VALSecureEnclaveValet.init(identifier: "UserPresence", accessControl: VALAccessControl.UserPresence) as VALSecureEnclaveValet!
+    let username : String = "CustomerPresentProof"
+
+
+    // MARK: Outlets
+
+    @IBAction func setOrUpdateItem(sender: UIResponder)
+    {
+        let stringToSet = "I am here! " + NSUUID.init().UUIDString
+        let setOrUpdatedItem = secureEnclaveValet.setString(stringToSet, forKey: username)
+        updateTextView([#function, (setOrUpdatedItem ? "Success" : "Failure")])
+    }
+
+    @IBAction func getItem(sender: UIResponder)
+    {
+        var userCancelled: ObjCBool = false
```

Comment:
According to the Swift Compiler ObjCBool is the representation of Objective-C BOOL in Swift because that underlying representation is different in 64 vs. 32-bit. If you want to represent it as Bool in swift, but pass it correctly to ObjC as an inout, you can do

```
let userCancelled: Bool = true
var userCancelledObjC = ObjCBool(userCancelled)
let password = secureEnclaveValet.stringForKey(username, userPrompt: "Use TouchID to retrieve password", userCancelled:&userCancelledObjC)
```

However, this is totally circumventing the point of passing an inout (pointer) to the function, considering that userCancelled was declared as let, it obviously cannot be modified by the function `secureEnclaveValet.stringForKey:userPrompt()`. However, if this isn't an interesting part of the function contract for this call site, it's not a problem.

I would approach the author of `secureEnclaveValet.stringForKey:userPrompt()` and ask why a BOOL pointer is necessary.